### PR TITLE
`$(tag)` template function

### PR DESCRIPTION
**Pull request by bazsi**
Syntax: 
```
    $(tag <name-of-the-tag> <value-if-set> <value-if-unset>)
```
 
Unless the value-if-set/unset arguments are specified $(tag) results in a
boolean type, expanding to "0" or "1" depending on whether the message has
the specified tag set.
    
If value-if-set/unset are present, $(tag) would return a string, picking the
second argument <value-if-set> if the message has <tag> and picking the
third argument <value-if-unset> if the message does not have <tag>

### CODE REVIEW

File: modules/basicfuncs/tf-tag.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+typedef struct _TFTagState
+{
+  LogTagId tag_id;
+  GString *value_if_set;
+  GString *value_if_unset;
```
Comment (alltilla) [inline modules/basicfuncs/tf-tag.c:28]: Don't we want to let the user use templates here?

File: modules/basicfuncs/tf-tag.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023 Balazs Scheidler <balazs.scheidler@axoflow.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
```
Comment (alltilla) [inline modules/basicfuncs/tf-tag.c:5]: I believe this should be LGPL.

File: tests/copyright/policy [hunk 3]
```diff
@@ -178,6 +178,7 @@ modules/affile/tests/test_file_writer\.c$
 modules/affile/stdout\.[ch]
 modules/afsocket/socket-options-unix\.[ch]$
 modules/basicfuncs/vp-funcs\.[ch]$
+modules/basicfuncs/tf-tag\.c$
```
Comment (alltilla) [inline tests/copyright/policy:181]: I believe this should go to the LGPL section.

Comment (bazsi) [reply tests/copyright/policy:181]: It was actually in the right section, but the license header was incorrect. I meant to license it under the GPL, as it's usual for modules.

Fixed this up.

File: modules/basicfuncs/tf-tag.c (hunk 1 above)
Comment (bazsi) [reply modules/basicfuncs/tf-tag.c:27]: I am not sure. If there's a usecase then both the tag name or the expanded values could become templates. But for now my intended use-case has been to convert a tag to a label in metrics-probe.

The value-if-set/unset are only there to make it easy to use a more descriptive value than 0 or 1, which this normally expands to.

So with this, I'd wait until a use-case emerges.

Comment (alltilla) [reply modules/basicfuncs/tf-tag.c:27]: It would be useful for AND-ing and OR-ing different tags.

Comment (alltilla) [reply modules/basicfuncs/tf-tag.c:27]: We can leave it as is for now, and extend it later, sure.

**Review by alltilla — APPROVED** [review]

**Pull request merged**